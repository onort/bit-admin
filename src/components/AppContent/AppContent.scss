@import "../../styles/main.scss";

.container {
  display: flex;
  flex-grow: 1;
  margin-left: $sidebar-width;
  min-height: calc(100vh - #{$appBar-height});
  transition: margin-left 0.3s;

  &.closed {
    margin-left: 0;
  }
}

.inner {
  flex-grow: 1;
  max-width: px-to-rem(1200px);
  margin: $spacer-md auto;
  padding: 0 $spacer-md;
}
