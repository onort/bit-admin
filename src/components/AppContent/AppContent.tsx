import React from "react"

import styles from "./AppContent.scss"

const AppContent: React.SFC = props => {
  return (
    <main className={styles.container}>
      <div className={styles.inner}>
        <p>This is content area.</p>
        {props.children}
      </div>
    </main>
  )
}

export default AppContent
