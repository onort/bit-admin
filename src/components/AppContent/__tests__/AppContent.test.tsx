import React from "react"
import { shallow } from "enzyme"

import AppContent from "../"

describe("<AppContent />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<AppContent sidebarOpen={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it("should add proper css class when sidebar is closed.", () => {
    const wrapper = shallow(<AppContent sidebarOpen={false} />)
    expect(wrapper.find("main").hasClass("closed")).toBe(true)
  })

  it("should render children correctly", () => {
    const testP = "My test paragraph."
    const wrapper = shallow(
      <AppContent sidebarOpen={true}>
        <p data-testid="testParagraph">{testP}</p>
      </AppContent>
    )
    expect(wrapper.find('[data-testid="testParagraph"]').text()).toBe(testP)
  })

  it("should be able to have custom classnames", () => {
    const wrapper = shallow(
      <AppContent className="myCustomClass" sidebarOpen={false} />
    )
    expect(wrapper.find("main").hasClass("myCustomClass")).toBe(true)
  })
})
