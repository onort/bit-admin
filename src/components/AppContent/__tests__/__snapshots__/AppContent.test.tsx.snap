// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContent /> should match snapshot 1`] = `
<main
  className="container"
>
  <div
    className="inner"
  >
    <p>
      This is content area.
    </p>
  </div>
</main>
`;
